src/components/registration/topicSelection/TopicSelection.tsx

```diff
@@ -45,6 +45,9 @@ export const TopicSelection: VFC<{
 	const [topics, setTopics] = useState<TopicsDataInterface[]>([]);
 	const [isLoading, setIsLoading] = useState<boolean>(false);
 	const [listView, setListView] = useState<boolean>(false);
+	const [topicGroupId, setTopicGroupId] = useState<number>(
+		sessionStorageRegistrationData.topicGroupId || undefined
+	);
 
 	const getTopic = (mainTopicId: number) => {
 		return topics?.filter((topic) => topic?.id === mainTopicId)?.[0];
```

Comment:
This is only a boyscout task, as it was there before, but could you change this to `topics?.find(...)` as `filter` always searches the whole array and only the first element gets pcked up